swim/node.go

```diff
@@ -296,27 +290,36 @@ func (n *Node) Ready() bool {
 //
 //= = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
 
-// BootstrapOptions provides options to bootstrap the node with
+// BootstrapOptions is a configuration struct passed to Node.Bootstrap.
 type BootstrapOptions struct {
-	// Slice of hosts to bootstrap with, prioritized over provided file
+	// The DiscoverProvider resolves a list of bootstrap hosts. If this is
+	// specified, it takes priority over the legacy Hosts and File options
+	// below.
+	DiscoverProvider DiscoverProvider
+
+	// Slice of hosts to bootstrap with, prioritized over provided file.
+	// TODO: Deprecate this option in favour of accepting only a DiscoverProvider.
 	Hosts []string
 
-	// File containing a JSON array of hosts to bootstrap with
+	// File containing a JSON array of hosts to bootstrap with.
+	// TODO: Deprecate this option in favour of accepting only a DiscoverProvider.
 	File string
 
-	// Whether or not gossip should start
+	// Whether or not gossip should be started immediately after a successful
+	// bootstrap.
 	Stopped bool
 
-	// Amount of time before join requests time out
+	// Amount of time before individual join requests time out.
 	JoinTimeout time.Duration
 
-	// Minimum number of nodes to join
+	// Minimum number of nodes to join to satisfy a bootstrap.
 	JoinSize int
 
-	// Maximum time to attempt joins before joining cluster times out
+	// Maximum time to attempt joins before the entire boostrap process times
+	// out.
 	MaxJoinDuration time.Duration
 
-	// number of nodes to attempt to join
+	// Number of nodes to attempt to join at once.
```

Comment:
This comment is slightly in correct insofar as it not matching what the Node.js version does. ParallelismFactor is actually a _multiple_ applied to `JoinSize` to determine the number of nodes that will receive a join request.